server/persistence/readclient/readclient_read.go

```diff
@@ -39,7 +39,33 @@ func (c *ReadClient) ReadUserWorkspace(
 		return kerrors.NewInternalError(err)
 	}
 
+	// apply is-owner label
+	ApplyIsOwnerLabel(r, user)
+
 	// return workspace
 	r.DeepCopyInto(obj)
 	return nil
 }
+
+func ApplyIsOwnerLabel(workspace *restworkspacesv1alpha1.Workspace, owner string) {
+	// do nothing on an empty workspace
+	if workspace == nil {
+		return
+	}
+
+	var value string
+
+	if workspace.Namespace == owner {
+		value = "true"
+	} else {
+		value = "false"
+	}
+
+	if workspace.Labels == nil {
+		workspace.Labels = map[string]string{
+			restworkspacesv1alpha1.LabelIsOwner: value,
+		}
+	} else {
+		workspace.Labels[restworkspacesv1alpha1.LabelIsOwner] = value
+	}
```

Comment:
I think we can simplify this a bit if you like it:
```suggestion
	if workspace.Labels == nil {
		workspace.Labels = map[string]string{}
	}

	switch workspace.Namespace {
	case owner:
		workspace.Labels[restworkspacesv1alpha1.LabelIsOwner] = "true"
	default:
		workspace.Labels[restworkspacesv1alpha1.LabelIsOwner] = "false"
	}
```

Reply:
added in a79432b